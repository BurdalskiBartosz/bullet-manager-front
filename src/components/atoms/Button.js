import React from 'react';
import { Button as MuiButton } from '@mui/material';

const Button = React.forwardRef(({ children, variant, ...rest }, ref) => {
	return (
		<MuiButton ref={ref} variant={variant} {...rest}>
			{children}
		</MuiButton>
	);
});

Button.displayName = 'Button';

export default Button;
