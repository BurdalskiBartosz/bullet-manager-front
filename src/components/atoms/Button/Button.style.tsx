import styled from 'styled-components';

export type tButtonColors = 'primary' | 'secondary' | 'dark';

type tButton = {
	fullWidth?: boolean;
	colorType?: tButtonColors;
};

export const StyledButton = styled.button<tButton>`
	width: ${({ fullWidth }) => (fullWidth ? '100%' : 'auto')};
	padding: 1rem 1.8rem;
	border: none;
	font-weight: 500;
	font-family: ${({ theme }) => theme.fonts.primary};
	color: ${({ theme }) => theme.colors.light};
	border-radius: ${({ theme }) => theme.sizes.borderRadius};
	cursor: pointer;
	transition: 0.3s linear;
	background-color: ${({ theme, colorType }) => {
		if (colorType === 'primary') {
			return theme.colors.purple100;
		} else if (colorType === 'secondary') {
			return theme.colors.purple200;
		} else {
			return theme.colors.purple500;
		}
	}};
`;
