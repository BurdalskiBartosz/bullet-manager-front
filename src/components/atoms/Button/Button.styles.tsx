import styled from 'styled-components';

export const StyledButton = styled.button`
	width: 100%;
	padding: 1rem 0;
	border: none;
	font-weight: bold;
	font-family: ${({ theme }) => theme.fonts.primary};
	color: ${({ theme }) => theme.colors.light};
	background-color: ${({ theme }) => theme.colors.primary};
	border-radius: ${({ theme }) => theme.sizes.borderRadius};
	cursor: pointer;
`;
