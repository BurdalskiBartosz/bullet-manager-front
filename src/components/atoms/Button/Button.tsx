import { FC } from 'react';
import { useTranslation } from 'react-i18next';
import { StyledButton, tButtonColors } from './Button.style';

type tProps = {
	children: string;
	fn?: Function;
	colorType?: tButtonColors;
};

const Button: FC<tProps> = ({
	children,
	colorType = 'primary',
	fn = () => {}
}) => {
	const { t } = useTranslation();
	const textContent = t(children);
	return (
		<StyledButton colorType={colorType} onClick={() => fn()}>
			{textContent}
		</StyledButton>
	);
};
export default Button;
