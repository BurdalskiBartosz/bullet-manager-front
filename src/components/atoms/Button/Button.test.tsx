import { render, screen } from '@testing-library/react';
import Button from './Button';

describe('renders learn react link', () => {
	it('should', () => {
		render(<Button>Test buttona</Button>);
		const linkElement = screen.getByText(/Test buttona/i);
		expect(linkElement).toBeInTheDocument();
	});
});
