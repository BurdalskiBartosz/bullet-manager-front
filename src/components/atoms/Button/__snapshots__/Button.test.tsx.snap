// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly without any props 1`] = `
.c0 {
  width: auto;
  padding: 1rem 1.8rem;
  border: none;
  font-weight: 500;
  font-family: Rubik,sans-serif;
  color: #FCFCFC;
  border-radius: 15px;
  cursor: pointer;
  -webkit-transition: 0.3s linear;
  transition: 0.3s linear;
  background-color: #6B48CE;
}

<button
  class="c0"
>
  Button
</button>
`;
